experimental/tests/features_test.py

```diff
@@ -0,0 +1,421 @@
+  def test_conv_features(self, W_std, b_std, n_layers, do_jit):
+    n, h, w, c = 3, 4, 5, 2
+    rng = random.PRNGKey(1)
+    x = self._get_init_data(rng, (n, h, w, c))
+
+    conv_args = {
+        'out_chan': 1,
+        'filter_shape': (3, 3),
+        'padding': 'SAME',
+        'W_std': W_std,
+        'b_std': b_std
+    }
+
+    kernel_fn = stax.serial(*[stax.Conv(**conv_args)] * n_layers)[2]
+    feature_fn = serial(*[ConvFeatures(**conv_args)] * n_layers)[1]
+
+    if do_jit:
+      kernel_fn = jit(kernel_fn)
+      feature_fn = jit(feature_fn)
+
+    k = kernel_fn(x)
+    f = feature_fn(x, [()] * n_layers)
+
+    k_nngp_approx = self._convert_image_feature_to_kernel(f.nngp_feat)
+    k_ntk_approx = self._convert_image_feature_to_kernel(f.ntk_feat)
+
+    self.assertAllClose(k.nngp, k_nngp_approx)
+    self.assertAllClose(k.ntk, k_ntk_approx)
+
+  @parameterized.named_parameters(
+      test_utils.cases_from_list({
+          'testcase_name':
+              ' [nlayers{}_{}] '.format(n_layers, 'jit' if do_jit else ''),
+          'n_layers':
+              n_layers,
+          'do_jit':
+              do_jit
+      } for n_layers in [1, 2, 3, 4] for do_jit in [True, False]))
+  def test_avgpool_features(self, n_layers, do_jit):
+    n, h, w, c = 3, 32, 28, 2
+    rng = random.PRNGKey(1)
+    x = self._get_init_data(rng, (n, h, w, c))
+
+    avgpool_args = {
+        'window_shape': (2, 2),
+        'strides': (2, 2),
+        'padding': 'SAME'
+    }
+
+    kernel_fn = stax.serial(*[stax.AvgPool(**avgpool_args)] * n_layers)[2]
+    feature_fn = serial(*[AvgPoolFeatures(**avgpool_args)] * n_layers)[1]
+
+    if do_jit:
+      kernel_fn = jit(kernel_fn)
+      feature_fn = jit(feature_fn)
+
+    k = kernel_fn(x)
+    f = feature_fn(x, [()] * n_layers)
+
+    k_nngp_approx = self._convert_image_feature_to_kernel(f.nngp_feat)
+
+    self.assertAllClose(k.nngp, k_nngp_approx)
+
+  def test_flatten_features(self):
+    n, h, w, c = 3, 32, 28, 2
+    n_layers = 1
+    rng = random.PRNGKey(1)
+    x = self._get_init_data(rng, (n, h, w, c))
+
+    kernel_fn = stax.serial(*[stax.Flatten()] * n_layers)[2]
+
+    k = kernel_fn(x)
+
+    feature_fn = serial(*[FlattenFeatures()] * n_layers)[1]
+
+    f = feature_fn(x, [()] * n_layers)
+
+    self.assertAllClose(k.nngp, f.nngp_feat @ f.nngp_feat.T)
+
+  @parameterized.named_parameters(
+      test_utils.cases_from_list({
+          'testcase_name':
+              ' [Wstd{}_bstd{}_depth{}_{}_{}] '.format(
+                  W_std, b_std, depth, relu_method, 'jit' if do_jit else ''),
+          'W_std':
+              W_std,
+          'b_std':
+              b_std,
+          'depth':
+              depth,
+          'relu_method':
+              relu_method,
+          'do_jit':
+              do_jit,
+      } for W_std in WEIGHT_VARIANCES for b_std in BIAS_VARIANCES
+                          for relu_method in ['PSRF'] for depth in [5]
+                          for do_jit in [False]))
```

Comment:
Could we test JIT as well?
```suggestion
                          for do_jit in [False, True]))
```